### PR TITLE
Adding tests: Tex{Sub}Image3D with web elements as input.

### PR DESCRIPTION
**Pull request by zhenyao**
This is intermediate work. May contains bugs.

### CODE REVIEW

File: sdk/tests/js/tests/tex-image-and-sub-image-utils.js [hunk 1]
```diff
@@ -99,6 +99,79 @@ var TexImageUtils = (function() {
...
+    '}'].join('\n');
+
+  /**
+   * A fragment shader for a single 2D_ARRAY texture.
+   * @type {string}
+   */
+  // Note we always use the first image in the array.
+  // Note we always output 1.0 for alpha because if the texture does not contain
+  // alpha channel, sampling returns 1; for RGBA textures, sampling returns [0,255].
+  var simple2DArrayTextureFragmentShaderES3 = [
```
Comment (kenrussell) [inline sdk/tests/js/tests/tex-image-and-sub-image-utils.js]: I think you meant to make this the uint variant.

File: sdk/tests/js/tests/tex-image-and-sub-image-utils.js [hunk 2]
```diff
@@ -127,6 +200,42 @@ var TexImageUtils = (function() {
...
+  var setupSimple2DArrayTextureFragmentShader = function(gl) {
+    return WebGLTestUtils.loadShader(gl, simple2DArrayTextureFragmentShaderES3, gl.FRAGMENT_SHADER);
+  };
+
+  /**
+   * Creates a simple 2D_ARRAY integer texture fragment shader.
+   * @param {!WebGLRenderingContext} gl The WebGLRenderingContext to use.
+   * @return {!WebGLShader}
+   */
+  var setupSimple2DArrayTextureFragmentShader = function(gl) {
```
Comment (kenrussell) [inline sdk/tests/js/tests/tex-image-and-sub-image-utils.js]: I think you meant to make this the uint variant.

**Comment by kenrussell** [comment]
LGTM overall. Couple of minor issues. Please merge yourself at your convenience after fixing them.

**Comment by zhenyao** [comment]
Updated accordingly

**Pull request merged**